ebos/eclwriter.hh

```diff
@@ -243,18 +250,23 @@ public:
             OPM_END_PARALLEL_TRY_CATCH("Collect to I/O rank: ",
                                        this->simulator_.vanguard().grid().comm());
         }
+        
 
         std::map<std::string, double> miscSummaryData;
         std::map<std::string, std::vector<double>> regionData;
         Inplace inplace;
+        
+        if (!isSubStep)
```

Comment:
I think you need to run this calculation even for substeps.  Otherwise, we'll risk breaking summary vectors like `FOIP` and `FOE`.